cmd/zrok/shareTui.go

```diff
@@ -211,6 +217,38 @@ func (m *shareModel) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
+func wrap(lines []string, width int) []string {
+	ret := make([]string, 0)
+	for _, line := range lines {
+		if width <= 0 || len(line) <= width {
+			ret = append(ret, line)
+			continue
+		}
+		for i := 0; i <= len(line); {
+			max := i + width
+			if max > len(line) {
+				max = len(line)
+			}
+			if line[i:max] == "" {
+				continue
+			}
+			nextI := i + width
+			if max < len(line)-1 {
+				if !wordwrapBreakpoints[rune(line[max])] || !wordwrapBreakpoints[rune(line[max+1])] {
+					lastSpace := strings.LastIndexAny(line[:max], wordwrapCharacters)
+					if lastSpace > -1 {
+						max = lastSpace
+						nextI = lastSpace
+					}
```

Comment:
My understanding is that this will do a hard wrap if a soft wrap is not possible, correct? We definitely want a forced hard wrap if the soft wrap fails... and this should fix the TUI formatting issues.

Reply:
Yes, if the word bleeds over past the line width it'll just hard break and go back to start from the word that broke it. It does kind of look funny when it's exactly the last word, but I think that's better than merging unrelated lines